gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java

```diff
@@ -64,10 +64,12 @@ public final class FlowCompilationValidationHelper {
    * For a given a flowSpec, verifies that an execution is allowed (in case there is an ongoing execution) and the
    * flowspec can be compiled. If the pre-conditions hold, then a JobExecutionPlan is constructed and returned to the
    * caller.
+   * @param flowSpec
+   * @param optionalFlowExecutionId provided for executions of scheduled events which should use a consistent id
    * @return jobExecutionPlan dag if one can be constructed for the given flowSpec
    */
-  public Optional<Dag<JobExecutionPlan>> createExecutionPlanIfValid(FlowSpec flowSpec)
-      throws IOException, InterruptedException {
+  public Optional<Dag<JobExecutionPlan>> createExecutionPlanIfValid(FlowSpec flowSpec,
+      Optional<String> optionalFlowExecutionId) throws IOException, InterruptedException {
```

Comment:
though not related to this PR's changes per se, just noting that guava lib upgrades cause us considerable hassle, and a special challenge we'd face if we ever elect to shade guava classes, would arise from using guava types as params or return types.

unless there's a pressing mandate to use guava `Optional`, we should always prefer `java.util.Optional`.  (guava has been deprecated in favor of that one for a very long time.)

Reply:
The other optionals used in this class are the guava one and reliant on intertwined `Orchestrator/DagManager` code.